int32 filter_size
